### PR TITLE
Run Service Catalog tests in CI

### LINKED ISSUE #410: Run Service Catalog integration tests in CI
In CI install Service Catalog into the Kind cluster and run integration tests against it.

### PR DESCRIPTION
**Pull request by ash2k**
Fixes #410.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,54 +7,76 @@ cache:
...
     - secure: "Hv6Sj29d9gI63bORCGHVeX5KqEryAAXf9INQL1ABs33a46wVzGCHM/9H/mE3FItHN7RZaeOEeUoDSePOt/8NBtZSAZ1PvbqhMgztvSWtweS9ezzClUHxN3BRZPSyNmZdrK0S9fo+NXgcgqcmOM9gQzrAduHIkWOOwdSGWgT6zNlAZovddwJ8RUtKavZ7U4szJyMtzlQ8S2K8PdTX9glH1U+8ZLHr54uDrhl3WqsCoBEV0GV59eOIaIseDorCEscVj1EDYCPdkYdU6/AWiuUlqXeOvC67CLcu/5W8m05YWXrEhKG+17jEWRPM+G+Kiyj6HccvjTTaXqTsx9ZLuWmSt86mgo15DmJlVtzpvkptgHhheDyp5MkifValXN3gvWzk3VqjHIUh81dP2o8iAWrGfJcYg+8hdv/obfNGvfcSKwgdefIp4B6f/VRvIySvHUqKavj2fEAGDMYHqahHyKGF/VegHuQbCT+F+mI+GdAwun7XzuDVAlftnbrl2kPq7Kjs+ahU4ZIU4pk6eY15rh/WPlR2v0M7O4nfv1IoyHh9F/G6foqwqDoT+RBF1VMr9WDEezz5B35si6ILI/+SG/7BnbHkFRGorOw86ZqMoIs7dkdpz8uHiz+ba83COuASbrHUMfTG3HQ65XY7puqccSPBQzg7yoVmqPUJJNkaroVMWmU="
 
-# Some of the instructions to install and run minikube are from https://blog.travis-ci.com/2017-10-26-running-kubernetes-on-travis-ci-with-minikube
 before_install:
+  # download things concurrently
+  - |
+    $(
```
Comment (ash2k) [inline .travis.yml]: Download all the things concurrently

File: .travis.yml [hunk 2]
```diff
@@ -7,54 +7,76 @@ cache:
...
-    docker pull kindest/node:${KUBE_VERSION}
-    kind create cluster --image="kindest/node:${KUBE_VERSION}"
 install:
   - make setup-ci
 
 script:
   - make update-bazel
   - make verify
   - make test-ci
+  - go get sigs.k8s.io/kind
```
Comment (ash2k) [inline .travis.yml]: This now runs after unit tests and verification - they are most common source of failures and also run reasonably fast.

File: .travis.yml [hunk 3]
```diff
@@ -7,54 +7,76 @@ cache:
...
+  - |
+    helm repo add svc-cat https://svc-catalog-charts.storage.googleapis.com
+    helm install svc-cat/catalog --name catalog --namespace catalog
+    helm install svc-cat/ups-broker --name ups-broker --namespace ups-broker
+  # build container and run checks while service catalog and UPS are starting
   - make docker
   - make check
+  - |
+    kubectl create -f it/sc/ups-clusterservicebroker.yaml
+    sleep 5
```
Comment (ash2k) [inline .travis.yml]: Give Service Catalog a few seconds to list classes from broker.

File: it/utils_for_tests.go [hunk 4]
```diff
@@ -255,6 +256,12 @@ func SetupApp(t *testing.T, bundle *smith_v1.Bundle, serviceCatalog, createBundl
 		cli := smithClient.SmithV1().Bundles(cfg.Namespace)
 		b, err := cli.Get(bundle.Name, meta_v1.GetOptions{})
 		if err == nil {
+			if t.Failed() {
```
Comment (ash2k) [inline it/utils_for_tests.go]: If an integration test fails, it's useful to see the current state of the bundle.

File: .travis.yml (hunk 3 above)
Comment (amckague) [reply .travis.yml]: add a polling loop like we have for cluster startup in setup_local_cluster

File: .travis.yml [hunk 5]
```diff
@@ -7,41 +7,48 @@ cache:
...
     chmod +x dep
     sudo mv dep /usr/local/bin/
   - |
-    curl -Lo install.sh "https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-installer-linux-x86_64.sh"
-    chmod +x install.sh
-    ./install.sh --user
-    rm -f install.sh
+    chmod +x install_bazel.sh
+    ./install_bazel.sh --user
+    rm -f install_bazel.sh
```
Comment (fcobb) [inline .travis.yml]: For all of these commands, do you need to chain them with `&`?

Comment (ash2k) [reply .travis.yml]: No, it works without them

**Pull request merged**